src/sfctl/auth.py

```diff
@@ -32,3 +32,24 @@ def signed_session(self):
             session.verify = False
 
         return session
+
+class AdalAuthentication(Authentication):
+    """Azure Active Directory authentication for Service Fabric clusters"""
+    access_token = None
+
+    def __init__(self, token, no_verify=False):
+        self.access_token = token
+        self.no_verify = no_verify
+
+    def signed_session(self):
```

Comment:
Right. But msrest doesn't work in CLI scenario, my plan is to support the initial AAD auth first while keep figuring out how to support refreshing meanwhile.